test/onnx/test_pytorch_onnx_no_runtime.py

```diff
@@ -453,8 +503,78 @@ def forward(self, input):
+        def bad_clamp(g, self, min, max):
+            return symbolic_helper._onnx_unsupported("Bad boy!")
+
+        class MyClip(torch.nn.Module):
+            def forward(self, x):
+                return torch.clamp(x, min=-0.5, max=0.5)
+
+        onnx_model = export_to_onnx(
+            MyClip(),
+            torch.randn(3, 4, requires_grad=True),
+            custom_ops=[common_utils.custom_op("aten::clamp", bad_clamp, 9)],
+            operator_export_type=torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK,
+        )
+        self.assertAtenOp(onnx_model, "clamp", "Tensor")
+
+    @common_utils.skipIfCaffe2
+    def test_clip_aten_fallback_explicit_request(self):
+        class MyClip(torch.nn.Module):
+            def forward(self, x):
+                return torch.clamp(x, min=-0.5, max=0.5)
+
+        def break_is_registered_op_api(opname, domain, version):
+            fake_missing_symbolics = ("clamp",)
+            if opname in fake_missing_symbolics:
+                return False
+            return (
+                (domain, version) in symbolic_registry._registry
+                and opname in symbolic_registry._registry[(domain, version)]
+            )
+
+        # Force missing symbolic for well-known op using a mock
+        onnx_model = export_to_onnx(
+            MyClip(),
+            torch.randn(3, 4, requires_grad=True),
+            mocks=[
+                unittest.mock.patch(
+                    "torch.onnx.symbolic_registry.is_registered_op",
+                    side_effect=break_is_registered_op_api,
+                )
+            ],
+            operator_export_type=torch.onnx.OperatorExportTypes.ONNX_ATEN_FALLBACK,
+        )
+        self.assertAtenOp(onnx_model, "clamp", "Tensor")
+
+    def _helper_test_to_(self, cast_fn: Callable[[torch.Tensor], torch.Tensor]):
+        """Helper to test aten::to(device) variants.
+
+        `cast_fn` is converted into a `torch.jit.script`. It wraps `aten::to`
+        during export to preventing the devices to be hard-coded.
+
+        Needed by detectron2 after https://github.com/facebookresearch/detectron2/pull/4132/
+        """
+        cast_fn = torch.jit.script(cast_fn)
+        onnx_model = export_to_onnx(cast_fn, torch.zeros([1, 3, 32, 32]))
+        for n in onnx_model.graph.node:
+            self.assertNotEqual(n.op_type, "To")
+            self.assertNotEqual(n.op_type, "Cast")
+
+    def test_to__cpu_string(self):
+        def cast_cpu_string(src: torch.Tensor) -> torch.Tensor:
+            return src.to("cpu")
+
+        self._helper_test_to_(cast_cpu_string)
+
+    def test_to__device_cpu_string(self):
+        def cast_device_cpu_string(src: torch.Tensor) -> torch.Tensor:
+            return src.to(device="cpu")
+
+        self._helper_test_to_(cast_device_cpu_string)
 
-common_utils.instantiate_parametrized_tests(TestOptionalOutput)
 
 if __name__ == "__main__":
-    unittest.main()
+    common_utils.run_tests()
```

Comment:
common_utils.run_tests() has more functionality than unittest.main() and also more informative error message if anything went off.